src/firefly/js/drawingLayers/NorthUpCompass.js

```diff
@@ -0,0 +1,174 @@
+import {MouseState} from '../visualize/VisMouseCntlr.js';
+import {PlotAttribute} from '../visualize/WebPlot.js';
+import CsysConverter from '../visualize/CsysConverter.js';
+import { makeOffsetPt, makeWorldPt, makeImagePt, makeViewPortPt,makeScreenPt} from '../visualize/Point.js';
+import BrowserInfo from '../util/BrowserInfo.js';
+import VisUtil from '../visualize/VisUtil.js';
+import ShapeDataObj from '../visualize/draw/ShapeDataObj.js';
+import {primePlot, getPlotViewById} from '../visualize/PlotViewUtil.js';
+import {getUIComponent} from './NorthUpCompassUI.jsx';
+import {makeFactoryDef} from '../visualize/draw/DrawLayerFactory.js';
+import {flux} from '../Firefly.js';
+import {makeDirectionArrowDrawObj} from '../visualize/draw/DirectionArrowDrawObj.js';
+
+const ID= 'NORTH_UP_COMPASS_TOOL';
+const TYPE_ID= 'NORTH_UP_COMPASS_TYPE';
+
+
+const selHelpText='North Arrow - EQ. J2000';
+const editHelpText='Help text here!';
+
+const factoryDef= makeFactoryDef(TYPE_ID,creator, getDrawData,/* getLayerChanges*/ null, onDetach,/*getUIComponent*/null);
+
+export default {factoryDef, TYPE_ID}; // every draw layer must default export with factoryDef and TYPE_ID
+
+
+var idCnt=0;
+
+
+/**
+ *
+ * @return {Function}
+ */
+function creator() {
+
+    var drawingDef= makeDrawingDef('red');
+    var actionTypes= [ImagePlotCntlr.UPDATE_VIEW_SIZE, ImagePlotCntlr.PROCESS_SCROLL];
+
+    idCnt++;
+
+    var options= {
+        hasPerPlotData:true,
+        isPointData:false,
+        canUserChangeColor: ColorChangeType.DYNAMIC
+    };
+    return DrawLayer.makeDrawLayer( `${ID}-${idCnt}`, TYPE_ID, 'North Arrow - EQ. J2000',
+                                     options, drawingDef, actionTypes);
+}
+
+function getDrawData(dataType, plotId, drawLayer, action, lastDataRet){
+    var drawCompass= makeCompass(plotId, action);
+    return drawCompass || lastDataRet;
+}
+
+function getLayerChanges(drawLayer, action) {
+
+    switch (action.type) {
+/*        case DrawLayerCntlr.ATTACH_LAYER_TO_PLOT:
+            return attach(drawLayer,action);
+        case DrawLayerCntlr.MODIFY_CUSTOM_FIELD:
+            console.log(action.type);
+            break;
+        case DrawLayerCntlr.FORCE_DRAW_LAYER_UPDATE:
+            console.log(action.type);
+            break;*/
+        case DrawLayerCntlr.PROCESS_SCROLL:
+            console.log(action.type);
+            break;
+
+    }
+    return null;
+
+}
+
+function overlay(action) {
+    /*var {plotIdAry}= action.payload;
+
+    var cc= CsysConverter.make(primePlot(visRoot(),plotIdAry[0]));
+    if (!cc) return null;
+    */
+    var {plotIdAry}= action.payload;
+    var drawCompass= makeCompass(plotIdAry[0], action);
+    return drawCompass;
+
+}
+
+//TODO: WHY and WHEN is it used????
+function onDetach(drawLayer,action) {
+    var {plotIdAry}= action.payload;
+    plotIdAry.forEach( (plotId) => dispatchAttributeChange(plotId,false,PlotAttribute.SHOW_COMPASS,null));
+}
```

Comment:
onDetach is not necessary for this drawing layer.  I is used for cleanup that has to happen outside of the DrawLayer object.